### ISSUE TITLE
Improve documentation about database migrations and decryption

### ISSUE THREAD

**Issue by PabloCastellano** [rotki/rotki#4726]
## Abstract

Hello. I have started working on #2526 and I'm having trouble finding developer documentation, specially about the database.

First, I would like to open my own database using something like sqlitebrowser in order to see existing values and help me better understand how rotki works (and debugging). However I have no idea how to decrypt it. I assume it is some standard sqlite feature but I haven't spent enough time reading the code.

It would be useful to have some guide about the database migrations too. How do they work and when are they needed. For example, is it needed if I am adding support for a new L2 network?

**Comment by PabloCastellano** [comment]
Hello @yabirgb! Thanks for your quick answer.

I have found out that sqlitebrowser in Ubuntu does not have sqlcipher support but I have workarounded it by following this [stackoverflow thread](https://stackoverflow.com/questions/48105035/sqlite-browser-without-sqlcipher-support-in-ubuntu). Now sqlitebrowser asks for a password and the sqlcipher version. From the `dbinfo.json` file I get that it is v4. However after entering my password, it looks like the password is wrong because it asks me again for the password and so on. Any idea on this?

Regarding your PR, I have read it and looks nice for future L2 networks support and I'm looking forward to see it merged. Since I'm playing around a bit with rotki I don't mind rebasing my PR later on.

Regarding the database migrations, I still don't know if I should include a database migration for Polygon support (probably because I still haven't figured out what is inside the encrypted database).

I have just joined the discord server, thanks!
